src/lib/bootloader/proposal_client.rb

```diff
@@ -226,7 +224,23 @@ def single_click_action(option, value)
 
       value ? stage1.add_udev_device(device) : stage1.remove_device(device)
 
+      update_required_packages
+
       Yast::Bootloader.proposed_cfg_changed = true
     end
+
+    def update_required_packages
+      bl = ::Bootloader::BootloaderFactory.current
+      bootloader_resolvables = Yast::PackagesProposal.GetResolvables("yast2-bootloader", :package)
+
+      if (bl.name == "none" || bl.stage1.devices.empty?) &&
+          !bootloader_resolvables.empty?
+        log.info "packages #{bl.packages} are no longer required"
+        Yast::PackagesProposal.RemoveResolvables("yast2-bootloader", :package, bl.packages)
```

Comment:
this is wrong. It should remove `bootloader_resolvables` and not bl.packages, which for `none` should be empty.